scalameta/tokenizers/shared/src/main/scala/scala/meta/internal/tokenizers/XmlParser.scala

```diff
@@ -11,124 +11,114 @@ import fastparse.all._
-    '\u3041' to '\u3094', '\u30A1' to '\u30FA', '\u3105' to '\u312C', '\uAC00' to '\uD7A3'
-  ))
-  val NameStartChar = P(CharIn(
-    ":", 'A' to 'Z', "_", 'a' to 'z', '\u00C0' to '\u00D6', '\u00D8' to '\u00F6',
-    '\u00F8' to '\u02FF', '\u0370' to '\u037D', '\u037F' to '\u1FFF', '\u200C' to '\u200D',
-    '\u2070' to '\u218F', '\u2C00' to '\u2FEF', '\u3001' to '\uD7FF', '\uF900' to '\uFDCF',
-    '\uFDF0' to '\uFFFD' // ++ [#x10000-#xEFFFF] ???? don't chars max out at \uffff ????
-  ))
-  val NameChar = P( NameStartChar | CharIn(
-    "-", ".", '0' to '9', "\u00B7", '\u0300' to '\u036F', '\u203F' to '\u2040'
-  ))
-  val Ideographic = P( CharIn(
-    '\u4E00' to '\u9FA5',  "\u3007", '\u3021' to '\u3029'
-  ))
-  // format: on
-  val Eq = P(WL.? ~ "=" ~ WL.?)
-  val ScalaExpr = P("{" ~ WS ~ Block ~ WL ~ "}")
-  val XNameStart = P("_" | BaseChar | Ideographic)
-  val Name = P(XNameStart ~ NameChar.rep)
-
-  val Char = P(AnyChar)
-  val Char1 = P(!("<" | "&") ~ Char)
-  val CharQ = P(!"\"" ~ Char1)
-  val CharA = P(!"'" ~ Char1)
-  val CharB = P(!"{" ~ Char1)
-
-  val Comment = P("<!--" ~ ((!"-" ~ Char) | ("-" ~ (!"-" ~ Char))).rep ~ "-->")
-
-  val EntityRef = P("&" ~ Name ~ ";")
-  val CharRef = P("&#" ~ CharIn('0' to '9').rep(1) ~ ";" | "&#x" ~ HexNum ~ ";")
-  val Reference = P(EntityRef | CharRef)
-  val AttValue = P(
-    "\"" ~ (CharQ | Reference).rep ~ "\"" |
-      "'" ~ (CharA | Reference).rep ~ "'" |
-      ScalaExpr
-  )
-  val Attribute = P(Name ~ Eq ~ AttValue)
-  val TagHeader = P("<" ~ Name ~ (WL ~ Attribute).rep ~ WL.?)
-  val CharData = P((!("{" | "]]>" | CharRef) ~ Char1 | "{{").rep(1))
-  val EmptyElemTagEnd = P("/>")
-  val STagEnd = P(">")
-
-  val CDStart = P("<![CDATA[")
-  val CData = P((!"]]>" ~ Char).rep)
-  val CDEnd = P("]]>")
-  val CDSect = P(CDStart ~ CData ~ CDEnd)
-
-  val PITarget = P(!(("X" | "x") ~ ("M" | "m") ~ ("L" | "l")) ~ Name)
-  val PI = P("<?" ~ PITarget ~ (WL ~ (!"?>" ~ Char).rep).? ~ "?>")
-
-  val ETagP = P("</" ~ Name ~ WL.? ~ ">")
-
-  val XmlContent: P0 = P(Element | CDSect | PI | Comment)
-  val Content1 = P(XmlContent | Reference | ScalaExpr)
-  val Content = P((CharData | Content1).rep)
-  val Element = P(TagHeader ~ (EmptyElemTagEnd | STagEnd ~ Content ~ ETag))
-
-  val ETag = P("</" ~ Name ~ WL.? ~ ">")
-
-  val TagPHeader = P("<" ~ Name ~ WL.?)
-  val EmptyElemTagPEnd = P("/>")
-  val STagPEnd = P(">")
-  val ScalaPatterns = P("{" ~ Patterns ~ WL ~ "}")
-  val ElemPattern: P0 = P(TagPHeader ~ (EmptyElemTagPEnd | STagPEnd ~ ContentP ~ ETagP))
-  val ContentP = P(CharData.? ~ ((ElemPattern | ScalaPatterns) ~ CharData.?).rep)
-  val ContentP1 = P(ElemPattern | Reference | CDSect | PI | Comment | ScalaPatterns)
-  val XmlExpr = P(WL ~ XmlContent ~ (WL ~ Element).rep)
-  val XmlPattern = P(WL ~ ElemPattern)
+                WL: P0 = CharsWhile(_.isWhitespace).opaque("whitespace")) {
+
+  val XmlExpr: P0 = P( Xml.XmlContent.rep(min = 1, sep = WL.?) )
+  val XmlPattern: P0 = P( Xml.ElemPattern )
```

Comment:
The xml parser is not the same in expression and pattern position. Your implementation always uses the parser for xml expression. It should work most of the time (except for some very weird cases) since xml in pattern position is a subset of xml in expression position